src/userinput.c

```diff
@@ -363,3 +363,31 @@ char *read_from_stdin(size_t count)
 	// Just keep using the larger buffer if realloc() fails.
 	return output ? output : buf;
 }
+
+/**
+ * Returns the given "input" prefixed with "prefix" in a dynamically
+ * allocated string. This behaves exactly like "strdup" if "input" already
+ * starts with "prefix".
+ */
+char *strdup_with_prefix(const char *input, const char *prefix)
+{
+	int input_len;
+	int prefix_len;
+	char *buf;
+
+	input_len = strlen(input);
+	prefix_len = strlen(prefix);
+
+	if (input_len >= prefix_len && memcmp(prefix, input, prefix_len) == 0)
+		return strdup(input);
+
+	buf = malloc(input_len + prefix_len);
+	if (buf == NULL)
+		return NULL;
+
+	memcpy(buf, prefix, prefix_len);
+	memcpy(buf + prefix_len, input, input_len);
+	buf[input_len + prefix_len] = '\0';
+
+	return buf;
+}
```

Comment:
I would rather add this as a static function in `src/main.c`. Or otherwise `src/config`.

Reply:
I turned it back into a static function, since I doubt it'll be needed elsewhere.